### ISSUE TITLE
Assertion failure on multibyte string - pg_verifymbstr

### ISSUE THREAD

**Issue by magi345** [greenplum-db/gpdb#8180]
### Greenplum version or build
PostgreSQL 9.4.20 (Greenplum Database 6.0.0-beta.4+dev.136.g6a02f289f4 build dev) on x86_64-apple-darwin18.6.0, compiled by Apple LLVM version 10.0.1 (clang-1001.0.46.4), 64-bit compiled on Jul 19 2019 12:31:53 (**with assert checking**)

### OS version and uname -a
Darwin Haos-MacBook-Pro.local 18.6.0 Darwin Kernel Version 18.6.0: Thu Apr 25 23:16:27 PDT 2019; root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64

### autoconf options used ( config.status --config )
'--prefix=/Users/haowang/usr/local/greenplum-db-6X' '--with-pgport=5432' '--with-libedit-preferred' '--with-pam' '--with-python' '--with-perl' '--enable-cassert' '--enable-debug' '--enable-depend' '--disable-mapreduce' '--disable-gpfdist' '--disable-orca' '--disable-gpperfmon' '--disable-gpcloud'

### Installation information ( pg_config )
BINDIR = /Users/haowang/usr/local/greenplum-db-6X/bin
DOCDIR = /Users/haowang/usr/local/greenplum-db-6X/share/doc/postgresql
HTMLDIR = /Users/haowang/usr/local/greenplum-db-6X/share/doc/postgresql
INCLUDEDIR = /Users/haowang/usr/local/greenplum-db-6X/include
PKGINCLUDEDIR = /Users/haowang/usr/local/greenplum-db-6X/include/postgresql
INCLUDEDIR-SERVER = /Users/haowang/usr/local/greenplum-db-6X/include/postgresql/server
LIBDIR = /Users/haowang/usr/local/greenplum-db-6X/lib
PKGLIBDIR = /Users/haowang/usr/local/greenplum-db-6X/lib/postgresql
LOCALEDIR = /Users/haowang/usr/local/greenplum-db-6X/share/locale
MANDIR = /Users/haowang/usr/local/greenplum-db-6X/share/man
SHAREDIR = /Users/haowang/usr/local/greenplum-db-6X/share/postgresql
SYSCONFDIR = /Users/haowang/usr/local/greenplum-db-6X/etc/postgresql
PGXS = /Users/haowang/usr/local/greenplum-db-6X/lib/postgresql/pgxs/src/makefiles/pgxs.mk
CONFIGURE = '--prefix=/Users/haowang/usr/local/greenplum-db-6X' '--with-pgport=5432' '--with-libedit-preferred' '--with-pam' '--with-python' '--with-perl' '--enable-cassert' '--enable-debug' '--enable-depend' '--disable-mapreduce' '--disable-gpfdist' '--disable-orca' '--disable-gpperfmon' '--disable-gpcloud'
CC = gcc
CPPFLAGS = -I/Users/haowang/usr/local/greenplum-db-6X/include
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -Wno-address -Wno-unused-command-line-argument -g -ggdb -O3 -std=gnu99 -Werror=uninitialized -Werror=implicit-function-declaration -I/Users/haowang/usr/local/greenplum-db-6X/include
CFLAGS_SL =
LDFLAGS = -Wl,-dead_strip_dylibs -L/Users/haowang/usr/local/greenplum-db-6X/lib
LDFLAGS_EX =
LDFLAGS_SL =
LIBS = -lpgcommon -lpgport -lpam -lzstd -lm -L/Users/haowang/usr/local/greenplum-db-6X/lib
VERSION = PostgreSQL 9.4.20

### Step to reproduce the behavior
Run attached SQL script
[long_err.sql.zip](https://github.com/greenplum-db/gpdb/files/3409478/long_err.sql.zip)

Will see following output
```
psql:long_err.sql:9: FATAL:  Unexpected internal error (elog.c:259)
DETAIL:  FailedAssertion("!(pg_verifymbstr(*str, len, 1))", File: "elog.c", Line: 259)
HINT:  Process 79058 will wait for gp_debug_linger=120 seconds before termination.
Note that its locks and other resources will not be released until then.
CONTEXT:  PL/Python function "long_err"
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
psql:long_err.sql:9: connection to server was lost
```

###In Addition###
There was a similar issue https://github.com/greenplum-db/gpdb/issues/7810
But it was closed without any discussion.

**Comment by hlinnaka** [comment]
So, the message gets truncated 1024 bytes. If the truncation happens in the middle of a multi-byte character, we trip the assertion in elog.c.

The assertion is GPDB specific. I'm not entirely sure why we've added it. PostgreSQL is more lenient in what it prints to the log. 

The python code was changed in PostgreSQL 9.6, so that it no longer truncates the error message (commit 5c3c3cd0a30). I don't think we should bother to do anything about this in GPDB, as it'll be changed once we merge with 9.6, anyway.

However, I'm not sure if there might be some other ways to trigger that assertion. Seems likely that there are more ways to get truncated strings into the log. But perhaps catching such cases was the point of the assertion in the first place.

In summary, I think we should just do nothing, and wait for 9.6. If you come with other ways to trigger the assertion, let's reconsider with the new example.

**Comment by hlinnaka** [comment]
PostgreSQL 9.6 has been merged to master, and I just tested that the repro script works on master now. 6X_STABLE still has the problem though.

**Comment by goopand** [comment]
@magi345 It doesn't look like an issue. Just reconfigure the source code without option '--enable-cassert' and it should work. Have a try.

**Comment by asimrp** [comment]
@goopand Turning off assertions is not the right way to address assertion failures :-). Did you get a chance to try the attached repro against Greenplum 6X?

**Comment by dpandhi-git** [comment]
main(7X) is fixed for 6X there is no information of problem existence.
